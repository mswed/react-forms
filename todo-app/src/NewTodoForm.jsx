import { useState } from 'react';
import { v4 as uuid } from 'uuid';

const NewTodoForm = ({ newTodo }) => {
  const initialState = { text: '' };
  const [formData, setFormData] = useState(initialState);
  const handleChange = (evt) => {
    const { name, value } = evt.target;
    setFormData((data) => ({ ...data, [name]: value }));
  };
  const handleSubmit = (evt) => {
    evt.preventDefault();
    newTodo({ text: formData.text, id: uuid() });
    setFormData(initialState);
  };
  return (
    <form onSubmit={handleSubmit}>
      <h2>New Todo</h2>
      <label htmlFor="text">Todo Text:</label>
      <input
        type="text"
        name="text"
        id="text"
        placeholder="Todo text"
        value={formData.text}
        onChange={handleChange}
      />
      <button>Create!</button>
    </form>
  );
};

export default NewTodoForm;
