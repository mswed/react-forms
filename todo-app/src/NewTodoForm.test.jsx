import { it, expect } from 'vitest';
import { render } from '@testing-library/react';
import NewTodoForm from './NewTodoForm';

it('Renders without crashing', async () => {
  render(<NewTodoForm />);
});

it('Matches the snapshot', async () => {
  const { asFragment } = render(<NewTodoForm />);
  expect(asFragment()).toMatchSnapshot();
});
