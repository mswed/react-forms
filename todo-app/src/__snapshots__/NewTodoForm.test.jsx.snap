// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Matches the snapshot 1`] = `
<DocumentFragment>
  <form>
    <h2>
      New Todo
    </h2>
    <label
      for="text"
    >
      Todo Text:
    </label>
    <input
      id="text"
      name="text"
      placeholder="Todo text"
      type="text"
      value=""
    />
    <button>
      Create!
    </button>
  </form>
</DocumentFragment>
`;
